### PR TITLE
Always remove package folder in conan create

### LINKED ISSUE #4804: package dir not cleaned when using conan create with -kb
When doing `conan create ... --keep-build`, the expectation is that the build directory will be left intact since the previous run, and the `package()` method will simply be re-run over that directory. That is happening successfully.  

However, when running `conan create... ` without `--keep-build` flag, the `package()` method always begins with a clean package directory as a destination for obvious reasons.  Conan implicitly clears the directory before running the `package()` method.  This "clearing" step is not currently happening when `--keep-build` flag is used.  It's easy to imagine how this detail was missed when `--keep-build` was implemented. Hopefully an easy fix. 

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: Bugfix: Remove package folder in ``conan create`` even when using ``--keep-build``
Docs: omit

- [x] Refer to the issue that supports this Pull Request: closes #4804 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: conans/test/functional/command/create_test.py [hunk 1]
```diff
@@ -68,29 +70,34 @@ def test(self):
...
+            from conans import ConanFile
+
+            class MyPkg(ConanFile):
+                def test(self):
+                    pass
+            """)
             client.save({"conanfile.py": conanfile,
                          "header.h": "",
                          "test_package/conanfile.py": test_conanfile})
+
```
Comment (jgsogo) [inline conans/test/functional/command/create_test.py]: The previous with/without test `if`-clause is in the wrong order

**Comment by danimtb** [comment]
Yes, you are totally right! Did not think about that. Thanks

**Comment by danimtb** [comment]
Requested changes done. I haven't added a test as the case fo cmake.install() in the build() method should be well covered in this test https://github.com/conan-io/conan/blob/develop/conans/test/functional/build_helpers/cmake_install_package_test.py#L46

**Comment by jgsogo** [comment]
But that test has a `client.run("remove * --force")`... maybe another one could cover this use case (I don't know how to look for it).

**Comment by danimtb** [comment]
That test should be enough to prove that the package folder is available in ``build()`` time. I have been thinking about the case of cmake.install() in build() together with ``conan create --keep-build`` and it wouldn't work either, as the build() is totally skipped when using ``--keep-build``. That is one of the reasons we recomend using it in the package(): https://docs.conan.io/en/latest/howtos/cmake_install.html (we should update and include the reasons there).

File: conans/client/installer.py [hunk 2]
```diff
@@ -227,6 +225,14 @@ def build_package(self, node, keep_build, recorder, remotes):
             return node.pref
 
 
+def remove_folder_raising(folder):
```
Comment (jgsogo) [inline conans/client/installer.py]: Make this function _private_, (`_remove_folder_raising`) or move it to `conans.client.tools.files`

File: conans/client/installer.py [hunk 3]
```diff
@@ -227,6 +225,14 @@ def build_package(self, node, keep_build, recorder, remotes):
...
+    try:
+        rmdir(folder)
+    except OSError as e:
+        raise ConanException("%s\n\nCouldn't remove folder, might be busy or open\n"
+                             "Close any app using it, and retry" % str(e))
```
Comment (jgsogo) [inline conans/client/installer.py]: Does the exception message include the folder itself? I would want to let the user know which one is the folder that might be in use by other app...

File: conans/client/installer.py [hunk 4]
```diff
@@ -190,10 +186,12 @@ def build_package(self, node, keep_build, recorder, remotes):
             with package_layout.conanfile_write_lock(self._output):
                 set_dirty(build_folder)
                 self._prepare_sources(conanfile, pref, package_layout, conanfile_path, source_folder,
-                                      build_folder, package_folder, remotes)
+                                      build_folder, remotes)
 
         # BUILD & PACKAGE
         with package_layout.conanfile_read_lock(self._output):
+            remove_folder_raising(package_folder)
+            mkdir(package_folder)
```
Comment (jgsogo) [inline conans/client/installer.py]: Is it needed? Sometimes we list the directories to know which packages are available (`PackageCacheLayout::conan_packages()`), maybe it is better to postpone the creation of any directory to the moment something is going to be copied there.

File: conans/test/functional/command/create_test.py [hunk 5]
```diff
@@ -135,6 +142,36 @@ class MyPkg(ConanFile):
...
+        package_files = os.listdir(client.cache.package(pref))
+        self.assertIn("header.h", package_files)
+        self.assertNotIn("source.cpp", package_files)
+        client.save({"conanfile.py": conanfile.replace("self.copy(\"*.h\")",
+                                                       "self.copy(\"*.cpp\")")})
+        client.run("create . pkg/0.1@danimtb/testing -kb")
+        package_files = os.listdir(client.cache.package(pref))
+        self.assertNotIn("header.h", package_files)
+        self.assertIn("source.cpp", package_files)
+
```
Comment (jgsogo) [inline conans/test/functional/command/create_test.py]: Do you think it is worth a test failing in the _build step_ and checking that the _package folder_ is not there?

File: conans/client/installer.py (hunk 3 above)
Comment (danimtb) [reply conans/client/installer.py]: Sure, the OSError exception includes the path of the folder

File: conans/test/functional/command/create_test.py (hunk 5 above)
Comment (danimtb) [reply conans/test/functional/command/create_test.py]: done

File: conans/client/installer.py (hunk 4 above)
Comment (danimtb) [reply conans/client/installer.py]: removed

File: conans/test/functional/command/create_test.py [hunk 6]
```diff
@@ -68,29 +70,34 @@ def test(self):
...
-            client.save({"conanfile.py": conanfile,
-                         "header.h": ""})
-        else:
-            test_conanfile = """from conans import ConanFile
-class MyPkg(ConanFile):
-    def test(self):
-        pass
-"""
+            test_conanfile = textwrap.dedent("""
+            from conans import ConanFile
```
Comment (memsharded) [inline conans/test/functional/command/create_test.py]: If using dedent, better really indent the contents one more indent, so it is not aligned with test_conanfile.

**Pull request merged**